pkg/workloads/manager.go

```diff
@@ -1169,3 +1164,36 @@ func (d *defaultManager) getRemoteWorkloadsFromState(
 
 	return remoteWorkloads, nil
 }
+
+// stopProcess stops the proxy process associated with the container
+func (d *defaultManager) stopProcess(ctx context.Context, name string) {
```

Comment:
Made this a private method of the workload manager, since no other part of the codebase should ever need to call this directly. It is assumed for now that this codepath will never be called in k8s. In future, we may want to refactor this interface so that the k8s proxy runner does not have a dependency on an interface which allows a workload to be stopped.